### PR TITLE
Add perftest implementation and run it as CI step

### LINKED ISSUE #100: performance testing
we need to develop some performance test, our actual implementation that is based on the single message immediate send path can be really slow, but need to be measured.

actually all other clients are able to send circa 1 mil/second messages with 3/4 bytes body.

the better approach is to run the java/C# perf test and comparing our numbers with theirs

### PR DESCRIPTION
**Pull request by icappello**
closes #100

### CODE REVIEW

File: src/consumer.ts [hunk 1]
```diff
@@ -3,7 +3,13 @@ import { Message } from "./producer"
 
 export type ConsumerFunc = (message: Message) => void
 
-export class Consumer {
+export interface IConsumer {
```
Comment (gpad) [inline src/consumer.ts:6]: Don't Use `I` pattern for interface

File: src/consumer.ts [hunk 2]
```diff
@@ -3,7 +3,13 @@ import { Message } from "./producer"
...
+  close(): Promise<void>
+  storeOffset(offsetValue: bigint): Promise<void>
+  queryOffset(): Promise<bigint>
+}
+
+export class Consumer implements IConsumer {
```
Comment (gpad) [inline src/consumer.ts:12]:
```suggestion
export class StreamConsumer implements Consumer {
```

File: src/index.ts [hunk 3]
```diff
@@ -1 +1,3 @@
 export * from "./connection"
+export { IProducer } from "./producer"
+export { IConsumer } from "./consumer"
```
Comment (gpad) [inline src/index.ts:3]: Also for producer

File: src/producer.ts [hunk 4]
```diff
@@ -48,12 +48,20 @@ interface MessageOptions {
...
   messageAnnotations?: Record<string, string | number>
 }
+
+export interface IProducer {
+  send(message: Buffer, opts?: MessageOptions): Promise<void>
+  sendSubEntries(messages: Message[], compressionType: CompressionType): Promise<void>
+  on(eventName: "publish_confirm", cb: PublishConfirmCallback): void
+  getLastPublishingId(): Promise<bigint>
+}
+
```
Comment (gpad) [inline src/producer.ts:58]:
```suggestion
export interface Producer {
  send(message: Buffer, opts?: MessageOptions): Promise<void>
  sendSubEntries(messages: Message[], compressionType: CompressionType): Promise<void>
  on(eventName: "publish_confirm", cb: PublishConfirmCallback): void
  getLastPublishingId(): Promise<bigint>
}

```

File: src/producer.ts [hunk 5]
```diff
@@ -48,12 +48,20 @@ interface MessageOptions {
...
 type PublishConfirmCallback = (err: number | null, publishingIds: bigint[]) => void
-export class Producer {
+export class Producer implements IProducer {
```
Comment (gpad) [inline src/producer.ts:60]:
```suggestion
export class StreamProducer implements IProducer {
```

File: performance_test/index.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    format.align(),
+    format.splat(),
+    format.label(),
+    format.printf((info) => `${info.timestamp} ${info.level}: ${info.message} ${info.meta ? inspect(info.meta) : ""}`)
+  ),
+  transports: new transports.Console(),
+})
+
+function parseArgs(args) {
+  const zip = (a, b) => Array.from(Array(Math.min(a.length, b.length)), (_, i) => [a[i], +b[i]])
```
Comment (tarzacodes) [inline performance_test/index.ts:22]: This is kinda hard to parse for me...
It's a bit more verbose, but maybe easier on the eyes?
const zip = <T, U>(a: T[], b: U[]): [T, U][] => {
  const shorterArray = a.length > b.length ? a : b
  const zipped = shorterArray.map((_, i) => [a[i], b[i]] as [T, U])
  return zipped
}

**Review by albertobarrila — APPROVED** [review]
LGTM
